docs/upstream_authentication.md

```diff
@@ -19,24 +19,56 @@ Combined, they allow stitch to authenticate to upstream data sources. Their sepa
 
 ## Upstream
 
-An Upstream resource answers the question "Which outgoing requests need which kind of authentication?".
+An Upstream resource modifies all outgoing request to particular remote host.
 
 Example:
 
 ```yaml
 kind: Upstream
 metadata:
-    name: organization-api
-    namespace: organizations-team
-host: 'organizations.example.com'
+  name: organization-api
+  namespace: organizations-team
+host: 'organizations.example.com' # deprecated
+sourceHosts:
+  - 'organizations.example.com'
+  - 'org.example.com'
+targetOrigin: https://org.dev.example.com
 auth:
-    type: ActiveDirectory
-    activeDirectory:
-        authority: <active directory authority url>
-        resource: <active directory resource id>
+  type: ActiveDirectory
+  activeDirectory:
+    authority: <active directory authority url>
+    resource: <active directory resource id>
+headers:
+  - name: some-header-name
+    value: '{incomingRequest?.headers?.["some-header-from-request"]'
 ```
 
-At runtime, `host` works as a selector - with this upstream applied, stitch knows that any request going out to `organizations.example.com` needs a `Bearer` token from the specified authority, for the specified resource.
+Properties:
+
+`host`: **Deprecated** Uses a selector for upstream. Use `sourceHosts` instead.
+
+`sourceHosts`: Uses as selector for upstream. If one of the values in list suits to url host the upstream is applied. Every value in list should be unique across all upstreams.
+
+> One (not both) of properties `host` or `sourceHosts` must be provided.
+
+`targetOrigin`: _(optional)_ Replaces the url origin.
+
+`auth`: _(optional)_ Allows to build `Authorization` header. For example using _Client Credentials_ flow.
+
+`headers`: _(optional)_ Allows to add or replace headers in outgoing request. Argument injection mechanism is available.
```

Comment:
The argument injection available here allows very different injection sources from the standard. We should detail the options here and mention that only these options are available.

I know the injection sources are in the examples but a short list with a brief description of each is still necessary (otherwise you always wonder if there are more not present in the examples that can be used)